src/lib/rnp.cpp

```diff
@@ -3433,7 +3433,7 @@ rnp_key_export(rnp_key_handle_t handle, rnp_output_t output, uint32_t flags)
     // handle flags
     if (flags & RNP_KEY_EXPORT_ARMORED) {
         flags &= ~RNP_KEY_EXPORT_ARMORED;
-        armored = true;
+        armored = !is_armored_dest(&output->dst);
```

Comment:
I don't think we should decide for the user what to do in this case. Say, what if he really wants to put some binary data into armored stream, and then add armored public key.